mscore/menus.cpp

```diff
@@ -992,40 +992,53 @@ void MuseScore::showPalette(bool visible)
       sp->setDrawGrid(true);
 
-      static const TempoPattern tp[] = {
-            TempoPattern("<sym>metNoteHalfUp</sym> = 80", 80.0/30.0),                    // 1/2
-            TempoPattern("<sym>metNoteQuarterUp</sym> = 80", 80.0/60.0),                 // 1/4
-            TempoPattern("<sym>metNote8thUp</sym> = 80", 80.0/120.0),                    // 1/8
-            TempoPattern("<sym>metNoteHalfUp</sym><sym>space</sym><sym>metAugmentationDot</sym> = 80", 120/30.0),       // dotted 1/2
-            TempoPattern("<sym>metNoteQuarterUp</sym><sym>space</sym><sym>metAugmentationDot</sym> = 80", 120/60.0),    // dotted 1/4
-            TempoPattern("<sym>metNote8thUp</sym><sym>space</sym><sym>metAugmentationDot</sym> = 80", 120/120.0),       // dotted 1/8
+      static const TempoPattern tps[] = {
+            TempoPattern("<sym>metNoteHalfUp</sym> = 80", 80.0/30.0, false),                    // 1/2
+            TempoPattern("<sym>metNoteQuarterUp</sym> = 80", 80.0/60.0, false),                 // 1/4
+            TempoPattern("<sym>metNote8thUp</sym> = 80", 80.0/120.0, false),                    // 1/8
+            TempoPattern("<sym>metNoteHalfUp</sym><sym>space</sym><sym>metAugmentationDot</sym> = 80", 120/30.0, false),       // dotted 1/2
+            TempoPattern("<sym>metNoteQuarterUp</sym><sym>space</sym><sym>metAugmentationDot</sym> = 80", 120/60.0, false),    // dotted 1/4
+            TempoPattern("<sym>metNote8thUp</sym><sym>space</sym><sym>metAugmentationDot</sym> = 80", 120/120.0, false),       // dotted 1/8
+            TempoPattern("<sym>metNoteQuarterUp</sym> = <sym>metNoteQuarterUp</sym><sym>space</sym><sym>metAugmentationDot</sym>", 3.0/2.0, true),
+            TempoPattern("<sym>metNoteQuarterUp</sym><sym>space</sym><sym>metAugmentationDot</sym> = <sym>metNoteQuarterUp</sym>", 2.0/3.0, true),
+            TempoPattern("<sym>metNoteHalfUp</sym> = <sym>metNoteQuarterUp</sym>", 1.0/2.0, true),
+            TempoPattern("<sym>metNoteQuarterUp</sym> = <sym>metNoteHalfUp</sym>", 2.0/1.0, true),
+            TempoPattern("<sym>metNoteHalfUp</sym> = <sym>metNoteHalfUp</sym><sym>space</sym><sym>metAugmentationDot</sym>", 3.0/2.0, true),
+            TempoPattern("<sym>metNoteHalfUp</sym><sym>space</sym><sym>metAugmentationDot</sym> = <sym>metNoteHalfUp</sym>", 2.0/3.0, true),
             };
-      for (unsigned i = 0; i < sizeof(tp)/sizeof(*tp); ++i) {
+      for (TempoPattern tp : tps) {
             TempoText* tt = new TempoText(gscore);
             tt->setFollowText(true);
-            // leave track at default (-1) to make it possible
-            // for drop() to tell that this came from palette
-            // (it will then be set to 0 there)
-            //tt->setTrack(0);
-            tt->setTempo(tp[i].f);
-            tt->setXmlText(tp[i].pattern);
-            sp->append(tt, tr("Tempo text"), QString(), 1.5);
+            tt->setXmlText(tp.pattern);
+            if (tp.relative) {
+                  if (basic) {
```

Comment:
maybe better doing `if (tp.relative && basic) continue;` at the very beginning of the loop, saves the delete and skips some unneeded function calls. Maybe even a `break', as long as we guarantee the relarive ones being after the absolute ones, we could save those extra loops